osu.Game/Screens/SelectV2/Carousel.cs

```diff
@@ -277,17 +277,27 @@ await Task.Run(async () =>
             void log(string text) => Logger.Log($"Carousel[op {cts.GetHashCode().ToString()}] {stopwatch.ElapsedMilliseconds} ms: {text}");
         }
 
-        private static void updateYPositions(IEnumerable<CarouselItem> carouselItems, float offset, float spacing)
+        private void updateYPositions(IEnumerable<CarouselItem> carouselItems, float offset)
         {
+            CarouselItem? previousVisible = null;
+
             foreach (var item in carouselItems)
+            {
+                float spacing = previousVisible == null || !item.IsVisible ? 0 : GetSpacingBetweenPanels(previousVisible, item);
                 updateItemYPosition(item, ref offset, spacing);
+
+                if (item.IsVisible)
+                    previousVisible = item;
+            }
         }
 
-        private static void updateItemYPosition(CarouselItem item, ref float offset, float spacing)
```

Comment:
I don't understand why this is not static.

Reply:
Simply because of the usage of `GetSpacingBetweenItems` inside the method, there's not an easy way to pass it along unless I pass it as a delegate, which sounds overkill.

Reply:
I can see usage in the method above, but not this one. Maybe it was removed in a refactor?

Reply:
Ah, that's right, but also while addressing https://github.com/ppy/osu/pull/31774#discussion_r1940576708 I've had to move the usage of that spacing method in here as well, so the static has to continue being removed.